doc/docs/master/how-tos/developer-workflow/build-pipelines.md

```diff
@@ -0,0 +1,120 @@
+
+Below is a Python example of a build pipline.
+
+```json
+{
+  "pipeline": {
+    "name": "map"
+  },
+  "description": "A pipeline that tokenizes scraped pages and appends counts of words to corresponding files.",
+  "transform": {
+    "build": {
+      "language": "python",
+      "path": "./source"
+    }
+  },
+  "input": {
+    "pfs": {
+      "repo": "scraper",
+      "glob": "/*"
+    }
+  }
+}
+```
+
+A build pipeline can be submitted the same way as any other pipeline, for example:
+
+```bash
+pachctl update pipeline -f <pipeline name>
+```
+
+## How it works
+
+When a build pipeline is submitted, the following actions occur:
+
+1. All files from the pipeline build `path` are copied to a PFS repository, `<pipeline name>_build`, which we can think of as the source code repository. In the case above, everything in `./source` would be copied to to the PFS `map_build` repository.
+
+2. A pipeline that uses the same repo but a different branch starts, reads the source code and creates build assets (i.e. pulling in dependencies and/or compiling) by running a `build.sh` script.
+
+3. The running pipeline, `<pipeline name>`, is updated with the the new source files and built assets then executes `sh /pfs/build/run.sh` when a job for that pipeline is started.
+
+!!! note
+      You can optionally specify a `.pachignore` file in the source root directory, which uses [ohmyglob](https://github.com/pachyderm/ohmyglob) entries to prevent certain files from getting pushed to this repo.
+
+The updated pipeline contains the following PFS repos mapped in as inputs:
+
+1. `/pfs/source` - source code that is required for running the pipeline.
+
+1. `/pfs/build` - any artifacts resulting from the build process.
+
+1. `/pfs/<input(s)>` - any inputs specified in the pipeline spec.
+
+## Builders
+
+The builder interprets the pipeline spec to determine:
+
+* A Docker image to use as the base image.
+* Steps to run for the build.
+* Step to run upon deployment.
+  
+The `transform.build.language` field is solely required to use an official builder (currently `python` or `go`), which already have impelmentations for `build.sh` and `run.sh`.
+
+### Python Builder
+
+The Python builder relies on a file structure similar to the following:
+
+```tree
+./map
+├── source
+│   ├── requirements.txt
+│   ├── ...
+│   └── main.py
+```
+There must exist a `main.py` which acts as the entrypoint for the pipeline. Optionally, a `requirements.txt` can be used to specify pip packages that will be installed during the build process. Other supporting files in the directory will also be copied and available in the pipeline if they are not excluded by the `.pachignore`.
+
+The `build.sh` and `run.sh` files are optional, as the Python Builder already contains these scripts. However, if the base image is modified, they must be provided.
```

Comment:
If I were reading this, I would still assume that it means I can optionally specify a ` build.sh` or `run.sh` in the `map/source` directory

Reply:
Ah you're right. Looks like I didn't commit that change. 

```suggestion
```